molecule/default/verify.yml

```diff
@@ -2,7 +2,35 @@
 - name: Verify role
   hosts: all
   become: yes
+  vars:
+    solr_core_conf_dest: "/var/solr/data/test/conf"
+    solr_config_xml_source: "/var/www/solr_conf/solrconfig.xml"
+    solr_core_properties_dest: "/var/solr/data/test/conf/solrconfig.xml"
 
   tasks:
+    - name: Check permissions on solr conf folder
+      shell: ls -ld "{{ solr_core_conf_dest }}" | grep 'solr'
+      register: solr_permission
+    - debug: var=solr_permission.stdout_lines
 
 
+    - name: Check the properties of source file
+      stat:
+        path: "{{ solr_config_xml_source }}"
+        checksum_algorithm: sha256
+      register: source
```

Comment:
@Chithrak07 to see why the assertion is failing, add debug tasks for both `source` and `dest` variables. Maybe something else is going on.

Reply:
The temp source directory didn't have the permissions set. I have set permissions and now it is running fine.